src/Type/Nette/StringsReplaceCallbackClosureTypeExtension.php

```diff
@@ -0,0 +1,97 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Nette;
+
+use Nette\Utils\Strings;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\StaticCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\Native\NativeParameterReflection;
+use PHPStan\Reflection\ParameterReflection;
+use PHPStan\TrinaryLogic;
+use PHPStan\Type\ClosureType;
+use PHPStan\Type\Constant\ConstantBooleanType;
+use PHPStan\Type\Constant\ConstantIntegerType;
+use PHPStan\Type\Php\RegexArrayShapeMatcher;
+use PHPStan\Type\StaticMethodParameterClosureTypeExtension;
+use PHPStan\Type\StringType;
+use PHPStan\Type\Type;
+use function array_key_exists;
+use const PREG_OFFSET_CAPTURE;
+use const PREG_UNMATCHED_AS_NULL;
+
+final class StringsReplaceCallbackClosureTypeExtension implements StaticMethodParameterClosureTypeExtension
+{
+
+	/** @var RegexArrayShapeMatcher */
+	private $regexArrayShapeMatcher;
+
+	public function __construct(RegexArrayShapeMatcher $regexArrayShapeMatcher)
+	{
+		$this->regexArrayShapeMatcher = $regexArrayShapeMatcher;
+	}
+
+	public function isStaticMethodSupported(MethodReflection $methodReflection, ParameterReflection $parameter): bool
+	{
+		return $methodReflection->getDeclaringClass()->getName() === Strings::class
+			&& $methodReflection->getName() === 'replace'
+			&& $parameter->getName() === 'replacement';
+	}
+
+	public function getTypeFromStaticMethodCall(MethodReflection $methodReflection, StaticCall $methodCall, ParameterReflection $parameter, Scope $scope): ?Type
+	{
+		$args = $methodCall->getArgs();
+		$patternArg = $args[1] ?? null;
+
+		if ($patternArg === null) {
+			return null;
+		}
+
+		$replacementType = $scope->getType($args[2]->value);
+
+		if (!$replacementType->isCallable()->yes()) {
+			return null;
+		}
+
+		$matchesType = $this->regexArrayShapeMatcher->matchExpr(
+			$patternArg->value,
+			$this->resolveFlagsType($args, $scope),
+			TrinaryLogic::createYes(),
+			$scope
+		);
+
+		if ($matchesType === null) {
+			return null;
+		}
+
+		return new ClosureType(
+			[
+				new NativeParameterReflection(
```

Comment:
Please create a new ParameterReflection class here instead of this and baseline :) You can even use an anonymous class.